apps/maestro/src/server/routers/sui/index.ts

```diff
@@ -215,16 +214,16 @@ export const suiRouter = router({
           throw new Error("Channel not found");
         }
 
-        if (minterAddress) {
+        if (minterAddress && !isCanonical) {
```

Comment:
Why do we need `isCanonical` check here given the coinManagement is always created with `new_with_cap` so it should always be `mint_burn`?